nevergrad/functions/powersystems/core.py

```diff
@@ -23,31 +22,31 @@ def __init__(self, input_size: int, output_size: int, layers: int = 3, layer_wid
         assert layers >= 2
         self.input_size = input_size
         self.output_size = output_size
-        self.layers: tp.List[tp.Any] = []
-        self.layers += [np.zeros((input_size, layer_width))]
+        self.layers = [np.zeros((layer_width, input_size))]
         for _ in range(layers - 2):
             self.layers += [np.zeros((layer_width, layer_width))]
-        self.layers += [np.zeros((layer_width, output_size))]
+        self.layers += [np.zeros((output_size, layer_width))]
         assert len(self.layers) == layers
 
     @property
     def dimension(self) -> int:
-        return sum([np.prod(l.shape) for l in self.layers])
-
-    def set_parameters(self, ww: tp.Any) -> None:
-        w = list(ww)
-        assert isinstance(w[0], float)
-        assert len(w) == self.dimension, f"length = {len(w)} instead of {self.dimension}: {w}."
-        for i in range(len(self.layers)):
-            s = np.prod(self.layers[i].shape)
-            self.layers[i] = np.reshape(np.array(w[:s]), self.layers[i].shape)  # TODO @oteytaud new name?
-            w = w[s:]
-
-    def get_output(self, inp: tp.Any) -> np.ndarray:
-        output = np.array(inp).reshape(1, len(inp))
+        return sum(layer.size for layer in self.layers)
+
+    def set_parameters(self, weights: np.ndarray) -> None:
+        if weights.size != self.dimension:
+            raise ValueError(f"length = {weights.size} instead of {self.dimension}: {weights}.")
+        start = 0
+        for i, layer in enumerate(self.layers):
+            numel = layer.size
+            self.layers[i] = weights[start: start + numel].reshape(layer.shape)
+            start += numel
+        if start != weights.size:
+            raise RuntimeError("Unexpected runtime error when distributing the weights")
+
+    def get_output(self, data: np.ndarray) -> np.ndarray:
         for l in self.layers[:-1]:
-            output = np.tanh(np.matmul(output, l))
-        return np.matmul(output, self.layers[-1])  # type: ignore
+            data = np.tanh(l @ data)
+        return self.layers[-1] @ data  # type: ignore
```

Comment:
matmul can be replaced with the @ operator in python 3. And with the shape update, this becomes a bit more natural and does not require reshaping.
It's also faster on my computer, but not on CircleCI so not sure it changed much in that aspect